src/views/Reports/ReportsModal/index.tsx

```diff
@@ -0,0 +1,355 @@
+import {
+    useCallback,
+    useState,
+} from 'react';
+import {
+    gql,
+    useMutation,
+} from '@apollo/client';
+import {
+    createSubmitHandler,
+    getErrorObject,
+    ObjectSchema,
+    requiredStringCondition,
+    useForm,
+} from '@togglecorp/toggle-form';
+import {
+    Button,
+    DateInput,
+    Modal,
+    SelectInput,
+    TextArea,
+    TextInput,
+} from '@togglecorp/toggle-ui';
+
+import FileInput from '#components/FileInput';
+import {
+    CreateReportInput,
+    CreateReportMutation,
+    CreateReportMutationVariables,
+    ReportTypeMutationResponseType,
+    StatusEnum,
+    UpdateReportInput,
+    UpdateReportMutation,
+    UpdateReportMutationVariables,
+} from '#generated/types/graphql';
+import useAlert from '#hooks/useAlert';
+import { transformToFormError } from '#utils/errorTransform';
+
+import styles from './styles.module.css';
+
+const CREATE_REPORT = gql`
+    mutation CreateReport($data: CreateReportInput!) {
+        createReport(data: $data) {
+            ... on ReportTypeMutationResponseType {
+                errors
+                ok
+                result {
+                    coverImage {
+                        name
+                        url
+                    }
+                    description
+                    id
+                    publishedDate
+                    reportFile {
+                        url
+                    }
+                    status
+                    title
+                }
+            }
+            ... on OperationInfo {
+                __typename
+                messages {
+                    message
+                }
+            }
+        }
+    }
+`;
+
+const UPDATE_REPORT = gql`
+    mutation UpdateReport($pk: ID!, $data: UpdateReportInput!) {
+        updateReport(pk: $pk, data: $data) {
+            ... on ReportTypeMutationResponseType {
+                errors
+                ok
+                result {
+                    description
+                    id
+                    publishedDate
+                    reportFile {
+                        url
+                    }
+                    status
+                    title
+                }
+            }
+            ... on OperationInfo {
+                __typename
+                messages {
+                    message
+                }
+            }
+        }
+    }
+`;
+
+interface Props {
+    onClose: () => void;
+    title: string;
+    initialValues?: Partial<UpdateReportInput & { id: string }>;
+    reportsRefetch: () => void;
```

Comment:
This naming convention is opposite of what we want. So, basically, this is called when the report is created or updated right? If that is the case then, the function should be called, onReportUpdate. ani outside you do the fetching separately. Please update it in other parts as well.